Assignment6/Array.cs

```
@@ -1,10 +1,71 @@
 ﻿using System;
+using System.Collections;
 using System.Collections.Generic;
+using System.Linq;
 
 namespace Assignment6
 {
-    public class Array<T>
+    public class Array<T> : ICollection<T>
     {
-        
+        private ICollection<T> _Collection;        
+
+        public int Count => _Collection.Count();        
+
+        public bool IsReadOnly => _Collection.IsReadOnly; 
+
+        public Array()
+        {
+            _Collection = new List<T>();
+        }
+
+        public Array(int newArraySize)
+        {
+            //          if int is < 0           false, do this                                   true, do this
+            _Collection = newArraySize < 0 ? new List<T>(newArraySize) : throw new ArgumentOutOfRangeException($"{nameof(newArraySize)} cannot be less than zero.");
+        }
+
+        public T this[int givenIndex]
+        {
+            get
+            {
+                ValidateIndexIsWithinArrayRange(givenIndex); //throws out of bounds exception
+                return _Collection.AsEnumerable().ElementAt(givenIndex);
+            }
+            set
+            {
+                ValidateIndexIsWithinArrayRange(givenIndex); //throws out of bounds exception
+                var temp = _Collection.ToList();
+                temp[givenIndex] = value;
+                _Collection = temp;
+            }
+        }        
+
+        public void Add(T addMe)
+        {          
+                _Collection.Add(addMe);            
+        }
+
+        public void Clear()
+        {
+            _Collection.Clear();            
+        }
+
+        public bool Contains(T checkMe) => _Collection.Contains(checkMe);
+
+        public void CopyTo(T[] array, int arrayIndex) => _Collection.CopyTo(array, arrayIndex);
+
+        public IEnumerator<T> GetEnumerator() => _Collection.GetEnumerator();
+
+        public bool Remove(T item) => _Collection.Remove(item);

```

I could be mistaken, but I believe Remove should throw an exception if an item does not exist

---

LinkedList Remove() in Java returns a boolean, and returns false if it can't find the item to remove. I don't know how Icollection does it, I haven't seen the source, but judging by the method signature Icollecion _should_ throw a false as well, which gets passed up the line. Since I'm using the array class basically as a wrapper for ICollection, I don't think altering the not found behavior for Remove is warranted.